### PR TITLE
Handle custom 2D coords correctly

### PR DESCRIPTION
**Pull request by lbdreyer**
This PR includes a few corrections needed for supporting plotting of 2D coords.

### CODE REVIEW

File: lib/iris/coords.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def __new__(cls, name_or_coord, minimum, maximum,
                                       'groupby_point, groupby_slice')
 
 
-def _discontinuity_in_2d_bounds(bds, abs_tol=1e-4):
+def _discontiguity_in_2d_bounds(bds, abs_tol=1e-4):
```
Comment (lbdreyer) [inline lib/iris/coords.py:146]: I've changed this to be `discontiguity` for consistency as we use the word `discontiguous` everywhere else

File: lib/iris/coords.py [hunk 2]
```diff
@@ -1066,14 +1066,15 @@ def is_contiguous(self, rtol=1e-05, atol=1e-08):
...
+                                         rtol=rtol, atol=atol)
             elif self.ndim == 2:
-                allclose, _, _ = _discontinuity_in_2d_bounds(self.bounds,
+                contiguous, _, _ = _discontiguity_in_2d_bounds(self.bounds,
                                                              abs_tol=atol)
-                return allclose
         else:
-            return False
+            contiguous = False
+        return contiguous
```
Comment (lbdreyer) [inline lib/iris/coords.py:1077]: Generally, multiple return statements are a bad thing

File: lib/iris/plot.py [hunk 3]
```diff
@@ -127,8 +126,6 @@ def get_span(coord):
     # Note the use of `reversed` to convert from the X-then-Y
     # convention of the end-user API to the V-then-U convention used by
     # the plotting routines.
-    coords = names(coords)
-
```
Comment (lbdreyer) [inline lib/iris/plot.py:131]: This was weird! I don't know why this was here. `names` effectively takes a list of coords and combines the names as a string, so
```names(['longitude', 'latitude'])```
would return the string `'longitude, latitude'` which would then mess up the processing below which expects coords to be a list.

File: lib/iris/plot.py [hunk 4]
```diff
@@ -344,16 +336,17 @@ def _draw_2d_from_bounds(draw_method_name, cube, *args, **kwargs):
     two_dim_contig_atol = kwargs.pop('two_dim_coord_contiguity_atol',
                                      1e-4)
     for coord in plot_defn.coords:
-        if coord.ndim == 2 and coord.has_bounds():
```
Comment (lbdreyer) [inline lib/iris/plot.py:347]: Check that `coord` is a coordinate first as coord could be either `None` or an integer

File: lib/iris/coords.py (hunk 1 above)
Comment (corinnebosley) [reply lib/iris/coords.py:146]: Agreed.  Why is it private though, is there a good reason?

File: lib/iris/plot.py (hunk 3 above)
Comment (corinnebosley) [reply lib/iris/plot.py:131]: I did this, and I thought I'd undone it.  This was an attempt to fix errors when coords are passed in as integers.

File: lib/iris/plot.py (hunk 4 above)
Comment (corinnebosley) [reply lib/iris/plot.py:347]: But now you're checking twice whether the coord has bounds or not.  One of the checks can be removed.

File: lib/iris/plot.py [hunk 5]
```diff
@@ -374,7 +367,7 @@ def _draw_2d_from_bounds(draw_method_name, cube, *args, **kwargs):
         string_axes = {}
 
         for coord, axis_name, data_dim in zip([u_coord, v_coord],
-                                              ['plotxaxis', 'yaxis'],
```
Comment (corinnebosley) [inline lib/iris/plot.py:377]: How weird.  Was this always 'plotxaxis' or did I change it accidentally?  I certainly don't remember doing that.

**Review by corinnebosley — CHANGES_REQUESTED** [review]

**Pull request merged**